src/plays/pokemon-stats/components/Search.tsx

```diff
@@ -1,6 +1,22 @@
 import React from "react";
 import "./search.css";
-const search = (props) => {
+type Propsi = {
+  setname: React.Dispatch<React.SetStateAction<string>>;
```

Comment:
@yung-coder , We should never pass the state action as a prop. we can create a handler like this and pass it as props.

```js
const handleSetName=(value:valueType)=>{
// set the state here 
}
```
Also change the `setname` prop to `onSetName`.